[indent-new-do-not-use] Error with ESLint 6.0.0 pre-releases

<!--
If you have a problem with a specific rule, please begin your issue title with [rulename] to make it easier to search for.
I.e. "[no-unused-vars] False positive when fooing the bar"

Please don't ignore this template.

If you ignore it, we're just going to respond asking you to fill it out, which wastes everyone's time.
The more relevant information you can include, the faster we can find the issue and fix it without asking you for more info.
-->

<!--
Are you opening an issue because the rule you're trying to use is not found?
🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨
1) Check the releases log: https://github.com/typescript-eslint/typescript-eslint/releases
    -  If the rule isn't listed there, then chances are it hasn't been released to the main npm tag yet.
2) Try installing the `canary` tag: `npm i @typescript-eslint/eslint-plugin@canary`.
    - The canary tag is built for every commit to master, so it contains the bleeding edge build. 
3) If ESLint still can't find the rule, then consider reporting an issue.
-->

**Repro**

<!--
Include a ***minimal*** reproduction case.
The more irrelevant code/config you give, the harder it is for us to investigate.
-->

```JSON
{
  "extends": "plugin:@typescript-eslint/eslint-recommended",
}
```

**Expected Result**

Able to parse and lint

**Actual Result**

Error: Failed to load plugin '@typescript-eslint': Cannot find module 'eslint/lib/util/ast-utils'

https://github.com/typescript-eslint/typescript-eslint/blob/cca1714c4bcb7a0349b4e10103870cc46bedf052/packages/eslint-plugin/src/rules/indent-new-do-not-use/index.ts#L11

**Additional Info**

```
Error: Failed to load plugin '@typescript-eslint' declared in '--config » ../../+typescript.js': Cannot find module 'eslint/lib/util/ast-utils'
Referenced from: /Users/teppei/src/github.com/teppeis/eslint-config-teppeis/+typescript.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
    at Function.Module._load (internal/modules/cjs/loader.js:508:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/@typescript-eslint/eslint-plugin/dist/rules/indent-new-do-not-use/index.js:7:21)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at /Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/requireindex/index.js:52:28
    at Array.forEach (<anonymous>)
    at Object.module.exports [as default] (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/requireindex/index.js:34:11)
    at Object.<anonymous> (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/@typescript-eslint/eslint-plugin/dist/index.js:9:37)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at ConfigArrayFactory._loadPlugin (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/config-array-factory.js:858:49)
    at ConfigArrayFactory._loadExtendedPluginConfig (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/config-array-factory.js:660:29)
    at ConfigArrayFactory._loadExtends (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/config-array-factory.js:606:29)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/config-array-factory.js:544:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/config-array-factory.js:490:20)
    at _normalizeObjectConfigData.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/config-array-factory.js:544:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/config-array-factory.js:490:20)
    at _normalizeObjectConfigData.next (<anonymous>)
    at createConfigArray (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/config-array-factory.js:306:25)
    at ConfigArrayFactory.loadFile (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/config-array-factory.js:381:16)
    at createCLIConfigArray (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:140:35)
    at new CascadingConfigArrayFactory (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:199:29)
    at new CLIEngine (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/node_modules/eslint/lib/cli-engine/cli-engine.js:552:36)
    at verify (/Users/teppei/src/github.com/teppeis/eslint-config-teppeis/test/index.js:23:18)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.9.0` |
| `@typescript-eslint/parser` | `1.9.0` |
| `TypeScript`                | `3.4.5` |
| `ESLint`                    | `6.0.0-alpha.2` |
| `node`                      | `10.15.3` |
| `npm`                       | `6.9.0` |

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -62,8 +62,6 @@ jobs:
           node_version: 10.x
         node_8_x:
           node_version: 8.x
-        node_6_x:
-          node_version: 6.x
     steps:
       - task: NodeTool@0
         inputs:

--- a/package.json
+++ b/package.json
@@ -64,7 +64,7 @@
     "all-contributors-cli": "^6.0.0",
     "babel-code-frame": "^6.26.0",
     "cz-conventional-changelog": "2.1.0",
-    "eslint": "^5.12.1",
+    "eslint": "^6.0.0",
     "eslint-plugin-eslint-plugin": "^2.0.1",
     "eslint-plugin-jest": "^22.2.2",
     "glob": "7.1.2",

--- a/packages/eslint-plugin-tslint/package.json
+++ b/packages/eslint-plugin-tslint/package.json
@@ -35,7 +35,7 @@
     "lodash.memoize": "^4.1.2"
   },
   "peerDependencies": {
-    "eslint": "^5.0.0",
+    "eslint": "^5.0.0 || ^6.0.0",
     "tslint": "^5.0.0"
   },
   "devDependencies": {

--- a/packages/eslint-plugin-tslint/tests/index.spec.ts
+++ b/packages/eslint-plugin-tslint/tests/index.spec.ts
@@ -1,4 +1,5 @@
 import { TSESLint } from '@typescript-eslint/experimental-utils';
+import * as parser from '@typescript-eslint/parser';
 import { readFileSync } from 'fs';
 import rule, { Options } from '../src/rules/config';
 
@@ -13,7 +14,7 @@ const ruleTester = new TSESLint.RuleTester({
      */
     project: './tests/tsconfig.json',
   },
-  parser: '@typescript-eslint/parser',
+  parser: require.resolve('@typescript-eslint/parser'),
 });
 
 /**
@@ -146,6 +147,7 @@ describe('tslint/error', () => {
   function testOutput(code: string, config: TSESLint.Linter.Config): void {
     const linter = new TSESLint.Linter();
     linter.defineRule('tslint/config', rule);
+    linter.defineParser('@typescript-eslint/parser', parser);
 
     expect(() => linter.verify(code, config)).toThrow(
       `You must provide a value for the "parserOptions.project" property for @typescript-eslint/parser`,
@@ -176,6 +178,7 @@ describe('tslint/error', () => {
     const linter = new TSESLint.Linter();
     jest.spyOn(console, 'warn').mockImplementation();
     linter.defineRule('tslint/config', rule);
+    linter.defineParser('@typescript-eslint/parser', parser);
     expect(() =>
       linter.verify('foo;', {
         parserOptions: {

--- a/packages/eslint-plugin/package.json
+++ b/packages/eslint-plugin/package.json
@@ -54,6 +54,6 @@
   },
   "peerDependencies": {
     "@typescript-eslint/parser": "^1.9.0",
-    "eslint": "^5.0.0"
+    "eslint": "^5.0.0 || ^6.0.0"
   }
 }

--- a/packages/eslint-plugin/src/rules/array-type.ts
+++ b/packages/eslint-plugin/src/rules/array-type.ts
@@ -72,7 +72,8 @@ function typeNeedsParentheses(node: TSESTree.Node): boolean {
   }
 }
 
-type Options = ['array' | 'generic' | 'array-simple'];
+export type OptionString = 'array' | 'generic' | 'array-simple';
+type Options = [OptionString];
 type MessageIds =
   | 'errorStringGeneric'
   | 'errorStringGenericSimple'

--- a/packages/eslint-plugin/src/rules/indent-new-do-not-use/index.ts
+++ b/packages/eslint-plugin/src/rules/indent-new-do-not-use/index.ts
@@ -8,7 +8,6 @@ import {
   AST_TOKEN_TYPES,
   TSESLint,
 } from '@typescript-eslint/experimental-utils';
-import { createGlobalLinebreakMatcher } from 'eslint/lib/util/ast-utils';
 import {
   isOpeningParenToken,
   isClosingParenToken,
@@ -25,6 +24,10 @@ import { TokenOrComment } from './BinarySearchTree';
 import { OffsetStorage } from './OffsetStorage';
 import { TokenInfo } from './TokenInfo';
 import { createRule, ExcludeKeys, RequireKeys } from '../../util';
+
+function createGlobalLinebreakMatcher() {
+  return /\r\n|[\r\n\u2028\u2029]/gu;
+}
 
 //------------------------------------------------------------------------------
 // Rule Definition

--- a/packages/eslint-plugin/tests/RuleTester.ts
+++ b/packages/eslint-plugin/tests/RuleTester.ts
@@ -8,13 +8,13 @@ const parser = '@typescript-eslint/parser';
 // but that's not as clean to type as this is
 type RuleTesterConfig = Omit<TSESLint.RuleTesterConfig, 'parser'> & {
   parser: typeof parser;
-}
+};
 class RuleTester extends TSESLint.RuleTester {
   constructor(options: RuleTesterConfig) {
     super({
       ...options,
       parser: require.resolve(options.parser),
-    })
+    });
   }
 
   run<TMessageIds extends string, TOptions extends Readonly<any[]>>(
@@ -29,12 +29,12 @@ class RuleTester extends TSESLint.RuleTester {
           throw new Error(errorMessage);
         }
       }
-    })
+    });
     tests.invalid.forEach(test => {
       if (test.parser === parser) {
         throw new Error(errorMessage);
       }
-    })
+    });
 
     super.run(name, rule, tests);
   }

--- a/packages/eslint-plugin/tests/RuleTester.ts
+++ b/packages/eslint-plugin/tests/RuleTester.ts
@@ -3,13 +3,13 @@ import * as path from 'path';
 
 const parser = '@typescript-eslint/parser';
 
-// re-export the RuleTester from here to make it easier to do the tests
-// as of eslint 6 you have to provide an absolute path to the parser
-// but that's not as clean to type as this is
 type RuleTesterConfig = Omit<TSESLint.RuleTesterConfig, 'parser'> & {
   parser: typeof parser;
 };
 class RuleTester extends TSESLint.RuleTester {
+  // as of eslint 6 you have to provide an absolute path to the parser
+  // but that's not as clean to type, this saves us trying to manually enforce
+  // that contributors require.resolve everything
   constructor(options: RuleTesterConfig) {
     super({
       ...options,
@@ -17,6 +17,9 @@ class RuleTester extends TSESLint.RuleTester {
     });
   }
 
+  // as of eslint 6 you have to provide an absolute path to the parser
+  // If you don't do that at the test level, the test will fail somewhat cryptically...
+  // This is a lot more explicit
   run<TMessageIds extends string, TOptions extends Readonly<any[]>>(
     name: string,
     rule: TSESLint.RuleModule<TMessageIds, TOptions>,

--- a/packages/eslint-plugin/tests/RuleTester.ts
+++ b/packages/eslint-plugin/tests/RuleTester.ts
@@ -1,8 +1,44 @@
 import { TSESLint, ESLintUtils } from '@typescript-eslint/experimental-utils';
 import * as path from 'path';
 
+const parser = '@typescript-eslint/parser';
+
 // re-export the RuleTester from here to make it easier to do the tests
-const RuleTester = TSESLint.RuleTester;
+// as of eslint 6 you have to provide an absolute path to the parser
+// but that's not as clean to type as this is
+type RuleTesterConfig = Omit<TSESLint.RuleTesterConfig, 'parser'> & {
+  parser: typeof parser;
+}
+class RuleTester extends TSESLint.RuleTester {
+  constructor(options: RuleTesterConfig) {
+    super({
+      ...options,
+      parser: require.resolve(options.parser),
+    })
+  }
+
+  run<TMessageIds extends string, TOptions extends Readonly<any[]>>(
+    name: string,
+    rule: TSESLint.RuleModule<TMessageIds, TOptions>,
+    tests: TSESLint.RunTests<TMessageIds, TOptions>,
+  ): void {
+    const errorMessage = `Do not set the parser at the test level unless you want to use a parser other than ${parser}`;
+    tests.valid.forEach(test => {
+      if (typeof test !== 'string') {
+        if (test.parser === parser) {
+          throw new Error(errorMessage);
+        }
+      }
+    })
+    tests.invalid.forEach(test => {
+      if (test.parser === parser) {
+        throw new Error(errorMessage);
+      }
+    })
+
+    super.run(name, rule, tests);
+  }
+}
 
 function getFixturesRootDir() {
   return path.join(process.cwd(), 'tests/fixtures/');

--- a/packages/eslint-plugin/tests/eslint-rules/no-redeclare.test.ts
+++ b/packages/eslint-plugin/tests/eslint-rules/no-redeclare.test.ts
@@ -1,5 +1,8 @@
 import rule from 'eslint/lib/rules/no-redeclare';
-import { AST_NODE_TYPES } from '@typescript-eslint/experimental-utils';
+import {
+  AST_NODE_TYPES,
+  AST_TOKEN_TYPES,
+} from '@typescript-eslint/experimental-utils';
 import { RuleTester } from '../RuleTester';
 
 const ruleTester = new RuleTester({
@@ -19,7 +22,6 @@ ruleTester.run('no-redeclare', rule, {
         ecmaVersion: 6,
       },
     },
-    'var Object = 0;',
     { code: 'var Object = 0;', options: [{ builtinGlobals: false }] },
     {
       code: 'var Object = 0;',
@@ -31,7 +33,11 @@ ruleTester.run('no-redeclare', rule, {
       options: [{ builtinGlobals: true }],
       parserOptions: { ecmaFeatures: { globalReturn: true } },
     },
-    { code: 'var top = 0;', env: { browser: true } },
+    {
+      code: 'var top = 0;',
+      env: { browser: true },
+      options: [{ builtinGlobals: false }],
+    },
     { code: 'var top = 0;', options: [{ builtinGlobals: true }] },
     {
       code: 'var top = 0;',
@@ -84,85 +90,115 @@ class D<T> {}
       parserOptions: { ecmaVersion: 6 },
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
       code: 'switch(foo) { case a: var b = 3;\ncase b: var b = 4}',
       errors: [
         {
-          message: "'b' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'b',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
       code: 'var a = 3; var a = 10;',
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
       code: 'var a = {}; var a = [];',
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
       code: 'var a; function a() {}',
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
       code: 'function a() {} function a() {}',
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
       code: 'var a = function() { }; var a = function() { }',
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
       code: 'var a = function() { }; var a = new Date();',
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
       code: 'var a = 3; var a = 10; var a = 15;',
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
@@ -170,9 +206,12 @@ class D<T> {}
       parserOptions: { sourceType: 'module' },
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
@@ -180,9 +219,12 @@ class D<T> {}
       parserOptions: { sourceType: 'module' },
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
@@ -190,9 +232,12 @@ class D<T> {}
       options: [{ builtinGlobals: true }],
       errors: [
         {
-          message: "'Object' is already defined.",
+          messageId: 'redeclaredAsBuiltin',
+          data: {
+            id: 'Object',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
@@ -200,9 +245,12 @@ class D<T> {}
       options: [{ builtinGlobals: true }],
       errors: [
         {
-          message: "'top' is already defined.",
+          messageId: 'redeclaredAsBuiltin',
+          data: {
+            id: 'top',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
       env: { browser: true },
     },
@@ -212,13 +260,19 @@ class D<T> {}
       parserOptions: { ecmaVersion: 6 },
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
         {
-          message: "'Object' is already defined.",
+          messageId: 'redeclaredAsBuiltin',
+          data: {
+            id: 'Object',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
@@ -227,9 +281,12 @@ class D<T> {}
       parserOptions: { ecmaVersion: 6, sourceType: 'module' },
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
@@ -238,9 +295,12 @@ class D<T> {}
       parserOptions: { ecmaVersion: 6, ecmaFeatures: { globalReturn: true } },
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
     {
@@ -249,9 +309,12 @@ class D<T> {}
       parserOptions: { ecmaVersion: 6 },
       errors: [
         {
-          message: "'a' is already defined.",
+          messageId: 'redeclared',
+          data: {
+            id: 'a',
+          },
           type: AST_NODE_TYPES.Identifier,
-        } as any,
+        },
       ],
     },
 
@@ -261,9 +324,12 @@ class D<T> {}
       options: [{ builtinGlobals: true }],
       errors: [
         {
-          message: "'b' is already defined.",
-          type: AST_NODE_TYPES.Identifier,
-        } as any,
+          messageId: 'redeclaredBySyntax',
+          data: {
+            id: 'b',
+          },
+          type: AST_TOKEN_TYPES.Block,
+        },
       ],
     },
   ],

--- a/packages/eslint-plugin/tests/rules/array-type.test.ts
+++ b/packages/eslint-plugin/tests/rules/array-type.test.ts
@@ -1,5 +1,6 @@
 import { TSESLint } from '@typescript-eslint/experimental-utils';
-import rule from '../../src/rules/array-type';
+import * as parser from '@typescript-eslint/parser';
+import rule, { OptionString } from '../../src/rules/array-type';
 import { RuleTester } from '../RuleTester';
 
 const ruleTester = new RuleTester({
@@ -886,25 +887,32 @@ let yyyy: Arr<Array<Array<Arr<string>>>> = [[[["2"]]]];`,
 // eslint rule tester is not working with multi-pass
 // https://github.com/eslint/eslint/issues/11187
 describe('array-type (nested)', () => {
-  describe('should deeply fix correctly', () => {
-    function testOutput(option: string, code: string, output: string): void {
-      it(code, () => {
-        const linter = new TSESLint.Linter();
+  const linter = new TSESLint.Linter();
+  linter.defineRule('array-type', rule);
+  linter.defineParser('@typescript-eslint/parser', parser);
 
-        linter.defineRule('array-type', Object.assign({}, rule) as any);
+  describe('should deeply fix correctly', () => {
+    function testOutput(
+      option: OptionString,
+      code: string,
+      output: string,
+    ): void {
+      it(code, () => {
         const result = linter.verifyAndFix(
           code,
           {
             rules: {
-              'array-type': [2, option],
+              'array-type': ['error', option],
             },
             parser: '@typescript-eslint/parser',
           },
           {
             fix: true,
-          },
+          }
         );
 
+        expect(result.messages).toHaveLength(0);
+        expect(result.fixed).toBeTruthy();
         expect(result.output).toBe(output);
       });
     }

--- a/packages/eslint-plugin/tests/rules/array-type.test.ts
+++ b/packages/eslint-plugin/tests/rules/array-type.test.ts
@@ -908,7 +908,7 @@ describe('array-type (nested)', () => {
           },
           {
             fix: true,
-          }
+          },
         );
 
         expect(result.messages).toHaveLength(0);

--- a/packages/eslint-plugin/tests/rules/ban-ts-ignore.test.ts
+++ b/packages/eslint-plugin/tests/rules/ban-ts-ignore.test.ts
@@ -52,7 +52,6 @@ if (false) {
   console.log("hello");
 }
             `,
-      parser: '@typescript-eslint/parser',
       errors: [
         {
           messageId: 'tsIgnoreComment',

--- a/packages/eslint-plugin/tests/rules/no-triple-slash-reference.test.ts
+++ b/packages/eslint-plugin/tests/rules/no-triple-slash-reference.test.ts
@@ -33,7 +33,6 @@ let a
 /// <reference path="Animal" />
 let a
             `,
-      parser: '@typescript-eslint/parser',
       errors: [
         {
           messageId: 'noTripleSlashReference',

--- a/packages/eslint-plugin/tests/rules/no-use-before-define.test.ts
+++ b/packages/eslint-plugin/tests/rules/no-use-before-define.test.ts
@@ -213,7 +213,6 @@ export namespace Third {
 }
             `,
       parserOptions: { ecmaVersion: 6, sourceType: 'module' },
-      parser: '@typescript-eslint/parser',
     },
     // https://github.com/eslint/typescript-eslint-parser/issues/550
     `
@@ -398,7 +397,7 @@ a();
     function a() {}
 }
             `,
-      parser: 'espree',
+      parser: require.resolve('espree'),
       errors: [
         {
           messageId: 'noUseBeforeDefine',

--- a/packages/eslint-plugin/tests/rules/type-annotation-spacing.test.ts
+++ b/packages/eslint-plugin/tests/rules/type-annotation-spacing.test.ts
@@ -1032,7 +1032,6 @@ type Bar = Record<keyof Foo, string>
           },
         },
       ],
-      parser: '@typescript-eslint/parser',
     },
     'let resolver: (() => PromiseLike<T>) | PromiseLike<T>;',
   ],
@@ -4432,7 +4431,6 @@ type Bar = Record<keyof Foo, string>
           },
         },
       ],
-      parser: '@typescript-eslint/parser',
     },
   ],
   invalid: [

--- a/packages/eslint-plugin/typings/eslint-ast-util.d.ts
+++ b/packages/eslint-plugin/typings/eslint-ast-util.d.ts
@@ -1,3 +1,0 @@
-declare module 'eslint/lib/util/ast-utils' {
-  export function createGlobalLinebreakMatcher(): RegExp;
-}

--- a/packages/eslint-plugin/typings/eslint-rules.d.ts
+++ b/packages/eslint-plugin/typings/eslint-rules.d.ts
@@ -212,7 +212,7 @@ declare module 'eslint/lib/rules/no-redeclare' {
   import { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
 
   const rule: TSESLint.RuleModule<
-    never,
+    'redeclared' | 'redeclaredAsBuiltin' | 'redeclaredBySyntax',
     [
       {
         builtinGlobals?: boolean;

--- a/packages/experimental-utils/src/ts-eslint/RuleTester.ts
+++ b/packages/experimental-utils/src/ts-eslint/RuleTester.ts
@@ -46,12 +46,8 @@ interface RunTests<
   invalid: InvalidTestCase<TMessageIds, TOptions>[];
 }
 interface RuleTesterConfig {
-  parser?:
-    | '@typescript-eslint/parser'
-    | 'espree'
-    | 'babel-eslint'
-    | 'esprima'
-    | string;
+  // should be require.resolve(parserPackageName)
+  parser: string;
   parserOptions?: ParserOptions;
 }
 declare interface RuleTester {

--- a/packages/parser/package.json
+++ b/packages/parser/package.json
@@ -38,7 +38,7 @@
     "typecheck": "tsc --noEmit"
   },
   "peerDependencies": {
-    "eslint": "^5.0.0"
+    "eslint": "^5.0.0 || ^6.0.0"
   },
   "dependencies": {
     "@types/eslint-visitor-keys": "^1.0.0",

--- a/packages/typescript-estree/tests/ast-alignment/utils.ts
+++ b/packages/typescript-estree/tests/ast-alignment/utils.ts
@@ -40,7 +40,7 @@ export function omitDeep(
     }
 
     for (const prop in node) {
-      if (node.hasOwnProperty(prop)) {
+      if (Object.prototype.hasOwnProperty.call(node, prop)) {
         if (shouldOmit(prop, node[prop])) {
           delete node[prop];
           continue;

